main/main.cpp

```diff
@@ -861,6 +861,10 @@ void Main::test_cleanup() {
 
 int Main::test_entrypoint(int argc, char *argv[], bool &tests_need_run) {
 	for (int x = 0; x < argc; x++) {
+		// Early return to ignore a possible user-provided "--test" argument.
+		if ((strlen(argv[x]) == 2) && ((strncmp(argv[x], "--", 2) == 0) || (strncmp(argv[x], "++", 2) == 0))) {
+			return EXIT_SUCCESS;
+		}
```

Comment:
Set `tests_need_run` to `false` as is done when not specifying `--test` near the end of the function:

```suggestion
		if ((strlen(argv[x]) == 2) && ((strncmp(argv[x], "--", 2) == 0) || (strncmp(argv[x], "++", 2) == 0))) {
			tests_need_run = false;
			return EXIT_SUCCESS;
		}
```